clang/lib/AST/PreorderAST.cpp

```diff
@@ -198,71 +188,71 @@ bool PreorderAST::CompareNodes(const Node *N1, const Node *N2) {
       return Lex.CompareExpr(L1->E, L2->E) == Result::LessThan;
     }
 
-    // N2:BinaryNodeExpr < N1:LeafExprNode.
+    // N2:OperatorNodeExpr < N1:LeafExprNode.
     return false;
   }
 
-  // N1:BinaryNodeExpr < N2:LeafExprNode.
+  // N1:OperatorNodeExpr < N2:LeafExprNode.
   if (isa<LeafExprNode>(N2))
     return true;
 
-  // Compare N1:BinaryNode and N2:BinaryNode.
-  const auto *B1 = dyn_cast<BinaryNode>(N1);
-  const auto *B2 = dyn_cast<BinaryNode>(N2);
+  // Compare N1:OperatorNode and N2:OperatorNode.
+  const auto *B1 = dyn_cast<OperatorNode>(N1);
+  const auto *B2 = dyn_cast<OperatorNode>(N2);
 
   if (B1->Opc != B2->Opc)
     return B1->Opc < B2->Opc;
   return B1->Children.size() < B2->Children.size();
```

Comment:
Minor observation (feel free to ignore if you wish): In almost all places variable `B` has been renamed to `O` except in the few lines above (variables `B1` and `B2`). There are two other similar occurrences of `B1` and `B2`.

Reply:
I have fixed these in my latest commit.